refactoringminer/src/main/java/gr/uom/java/xmi/LocationInfo.java

```diff
@@ -1,41 +1,65 @@
 package gr.uom.java.xmi;
 
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiJavaCodeReferenceElement;
+import com.intellij.psi.PsiVariable;
 import gr.uom.java.xmi.diff.CodeRange;
-import org.eclipse.jdt.core.dom.ASTNode;
-import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.jetbrains.annotations.NotNull;
 
+/**
+ * Provides an information about the element's location in the file.
+ */
 public class LocationInfo {
     private final String filePath;
     private final int startOffset;
     private final int endOffset;
-    private final int length;
     private final int startLine;
     private final int endLine;
+    private final int startColumn;
+    private final int endColumn;
     private final CodeElementType codeElementType;
-    private int startColumn;
-    private int endColumn;
 
-    public LocationInfo(CompilationUnit cu, String filePath, ASTNode node, CodeElementType codeElementType) {
+    public LocationInfo(@NotNull PsiFile file, @NotNull String filePath, @NotNull PsiElement node,
+                        @NotNull CodeElementType codeElementType) {
         this.filePath = filePath;
         this.codeElementType = codeElementType;
-        this.startOffset = node.getStartPosition();
-        this.length = node.getLength();
-        this.endOffset = startOffset + length;
-
-        //lines are 1-based
-        this.startLine = cu.getLineNumber(startOffset);
-        this.endLine = cu.getLineNumber(endOffset);
-        //columns are 0-based
-        this.startColumn = cu.getColumnNumber(startOffset);
-        //convert to 1-based
-        if (this.startColumn > 0) {
-            this.startColumn += 1;
+
+        TextRange range = getEclipseRange(node);
+        this.startOffset = range.getStartOffset();
+        this.endOffset = range.getEndOffset();
+
+        Document document = file.getViewProvider().getDocument();
+        if (document != null) {
+            this.startLine = document.getLineNumber(startOffset) + 1;
+            this.endLine = document.getLineNumber(endOffset) + 1;
+            this.startColumn = startOffset - document.getLineStartOffset(startLine - 1) + 1;
+            this.endColumn = endOffset - document.getLineStartOffset(endLine - 1) + 1;
+        } else {
+            this.startLine = 0;
+            this.endLine = 0;
+            this.startColumn = 0;
+            this.endColumn = 0;
         }
-        this.endColumn = cu.getColumnNumber(endOffset);
-        //convert to 1-based
-        if (this.endColumn > 0) {
-            this.endColumn += 1;
+    }
+
+    public static TextRange getEclipseRange(@NotNull PsiElement node) {
```

Comment:
do we need it only for comparing with the original tool?

Reply:
Hmm. You say that ranges matters in some metrics and I don't know in what cases it's important. If it's not important most of places, I'll concentrate on other differences. I planned to do as eclipse do in the and when I'll finish everything, remove cases which not affect any metrics

Reply:
add todo about removing this method when migration if finished